### ISSUE TITLE
`undici.fetch` fails (terminates) on 3XX  with `redirect: "manual"`

### ISSUE THREAD

**Issue by AlttiRi** [nodejs/undici#1622]
## Bug Description

`undici.fetch` fails on 301 "Moved Permanently" (and similar redirects) with `redirect: "manual"`

## Reproducible By

```js
import {fetch} from "undici";
// import fetch from "node-fetch"; // works fine

const url = "https://tinyurl.com/2af6by3e"; // 301 Moved Permanently
// const url = "https://tinyurl.com/y3rv756e"; // 302 Found

const resp = await fetch(url, {
    redirect: "manual"
});

console.log(resp.status, resp.statusText, resp.redirected);
console.log(resp.url);
console.log(resp.headers.get("location"));
console.log(resp);

const text = await resp.text();
console.log("---");
console.log(text);
console.log("---");

```

## Expected Behavior

```
301 Moved Permanently false
https://tinyurl.com/2af6by3e
https://github.com/nodejs/undici/issues/1622
Response {
...
}
---
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <meta http-equiv="refresh" content="0;url='https://github.com/nodejs/undici/issues/1622'" />

        <title>Redirecting to https://github.com/nodejs/undici/issues/1622</title>
    </head>
    <body>
        Redirecting to <a href="https://github.com/nodejs/undici/issues/1622">https://github.com/nodejs/undici/issues/1622</a>.
    </body>
</html>
---
```

## Result

```
301 Moved Permanently false
https://tinyurl.com/2af6by3e
https://github.com/nodejs/undici/issues/1622
Response {
...
}

TypeError: terminated
at Fetch.onAborted (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:1890:49)
at Fetch.emit (node:events:513:28)
at Fetch.terminate (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:83:10)
at Object.onError (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:2027:34)
at Request.onError (C:\Dev\downloader\node_modules\undici\lib\core\request.js:265:27)
at errorRequest (C:\Dev\downloader\node_modules\undici\lib\client.js:1722:13)
at Object.abort (C:\Dev\downloader\node_modules\undici\lib\client.js:1335:7)
at Object.destroy (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:1553:21)
at httpFetch (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:1051:39)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    [cause]: DOMException [AbortError]: The operation was aborted.
        at new DOMException (node:internal/per_context/domexception:53:5)
    at Object.destroy (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:1553:29)
    at httpFetch (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:1051:39)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async schemeFetch (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:875:14)
    at async C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:570:16
    at async mainFetch (C:\Dev\downloader\node_modules\undici\lib\fetch\index.js:554:16)
}

Node.js v18.7.0
```

## Environment

- "undici": "5.10.0"

**Comment by KhafraDev** [comment]
See #1618

**Comment by KhafraDev** [comment]
Fixed in #1627
